applications/machine_learning/Kconfig

```diff
@@ -5,6 +5,7 @@
 #
 
 menu "Machine learning application"
+rsource "Kconfig.ml_app"
```

Comment:
I have some doubts if this makes some real sense to move the main application settings to external file. I would rather expect for all this settings to be placed here. Look into serial_lte_modem, matter_weather_station or both assert trackers.

Reply:
I did it on purpose. In my opinion this:
![Screenshot from 2022-01-19 11-36-55](https://user-images.githubusercontent.com/52450830/150113918-016d8d59-34bf-475b-9c2d-f78a16fe8802.png)
and if you enter `Machine Learning Application`
![Screenshot from 2022-01-19 11-37-07](https://user-images.githubusercontent.com/52450830/150113913-377d8494-0fbd-47f7-956a-69509736c7fa.png)

looks much better and is logical. Application related options belong to `Machine Learning Application` submenu.
If i did it like you describe we would have some selectable options at first menuconfig view. I believe the intention was to split into App specific and Zephyr kernel specific options and i didn't want to break this.